### PR TITLE
#41 - Create PHPUnit tests

### PR DESCRIPTION
**Pull request by svendecabooter**
This PR contains PHPUnit tests for the classes in Encrypt where it's useful to create a unit test.

### CODE REVIEW

**Comment by svendecabooter** [comment]
I updated this pull request with a refactored version of the EncryptService class.

Changes:
* The encrypt() and decrypt() methods now have an EncryptionProfile entity as second parameter. This is more best practice for dependency injection, and allows for type hinting to enforce a valid EncryptionProfile to work with.
* This means the API has changed a bit, but I don't think there are any consumers yet. Modules using the EncryptService service will probably already have loaded the appropriate EncryptionProfile anyway, so they can just pass it along to the service.
* The refactoring also makes sure the duplicated code in the encrypt() and decrypt() methods is reduced to a minimum - which should improve maintainability.
* Instead of throwing a generic Exception object, the service now throws an EncryptException, which should improve debugging when things go wrong, and is consistent with how Drupal core throws exceptions.
* The PHPUnit tests have been updated to reflect these changes.

The Travis CI bot seems to throw errors, but I think this is unrelated to my code. The unit tests all pass locally. Maybe the Simpletests in the module are failing, but that should be covered in another pull request I reckon.

Thanks for your feedback!

File: readme.md [hunk 1]
```diff
@@ -20,12 +20,10 @@ It requires a key, which is provided by the Key module. To manage keys, visit `a
 After configuring the service, the service provides the ability to encrypt and decrypt.
 
 ### Encrypt
-
-`Drupal::service('encryption')->encrypt($string);`
+`$encryption_profile = \Drupal::service('entity.manager')->getStorage('encryption_profile')->load($instance_id);`
```
Comment (nerdstein) [inline readme.md]: I like putting this in, it's a bit more intuitive, as it shows what the second parameter is. We should consider adding in $string as well (I know it sounds silly).

I hope it doesn't confuse anyone, the code is not really commented or anything.

**Comment by nerdstein** [comment]
There is a lot going on here. Overall, I thought it was good. The service is further decoupled, which is fine and probably is better OO practices. I'm going to merge.

@svendecabooter - I cannot merge this due to a conflict with readme.md. Can you please address the conflict? I will merge there after

**Comment by nerdstein** [comment]
Merging

**Pull request merged**